.github/workflows/image_build_push.yml

```diff
@@ -0,0 +1,144 @@
+
+jobs:
+  fetch_run_id:
+    runs-on: ubuntu-latest
+
+    outputs:
+      run_id: ${{ steps.get_run_id.outputs.run_id }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with: 
+          python-version: '3.x'
+
+      - name: Install Python dependencies
+        run: |
+         python -m pip install --upgrade pip
+         pip install requests
+
+      - name: Run Python script
+        id: run_script
+        run: |
+          echo "Fetching latest successful run ID from e-misison-server docker image workflow"
+          python .github/fetch_runID.py
+    
+      - name: Get Run ID
+        id: get_run_id
+        run: echo "run_id=${{ steps.run_script.outputs.run_id }}" >> "$GITHUB_OUTPUT"
+
+  fetch_tag:
+      needs: fetch_run_id
+      runs-on: ubuntu-latest
+
+      env:
+        RUN_ID: ${{ needs.fetch_run_id.outputs.run_id }}
+
+      outputs:
+        docker_image_tag: ${{ steps.get_docker_tag.outputs.docker_image_tag }}
+
+      steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Run ID from previous fetch_run_id job
+        run: echo Run ID from previous job ${{ env.RUN_ID }}
+        
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          # TODO: Create a token with basic repo permissions
+          name: docker-image-tag
+          github-token: ${{ secrets.GH_PAT_TAG }}
+          repository: e-mission/e-mission-server
+          run-id: ${{ env.RUN_ID }}
+
+      - name: Print artifact tag
+        id: get_docker_tag
+        run: |
+          cat tag_file.txt
+          docker_image_tag=$(cat tag_file.txt)
+          echo $docker_image_tag
+          echo "docker_image_tag=$(echo $docker_image_tag)" >> $GITHUB_OUTPUT
+  build:
+    needs: fetch_tag
+
+    runs-on: ubuntu-latest
+
+    env:
+      DOCKER_IMAGE_TAG_1: ${{ needs.fetch_tag.outputs.docker_image_tag }}
+      DOCKER_IMAGE_TAG_2: ${{ github.event.inputs.docker_image_tag }}
+      
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Print input docker image tag
+      run: |
+        echo "Event name: ${{ github.event_name }}"
+        echo "Latest docker image tag (push): ${{ env.DOCKER_IMAGE_TAG_1 }}"
+        echo "Latest docker image tag (workflow_dispatch): ${{ env.DOCKER_IMAGE_TAG_2 }}"
+
+    - name: Update .env file 
+      run: |
+        if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+          echo "Workflow_dispatch: New server image built and pushed, Updating image tag in .env"
+          echo "DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_2" > .env
+        else
+          echo "Push event: Restoring latest server image tag in .env"
+          echo "DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_1" > .env
+        fi
+
+    - name: Add, Commit, Push changes to .env file
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
```

Comment:
nit, future fix: configure this to be more meaningful; "GitHub Action" doing what?